include/openthread/dnssd_server.h

```diff
@@ -147,20 +147,33 @@ typedef enum
     OT_DNSSD_QUERY_TYPE_RESOLVE_HOST = 3, ///< Service type resolve hostname.
 } otDnssdQueryType;
 
+/**
+ * Includes the count of queries, responses, failures handled by upstream DNS server.
```

Comment:
Let's add comment that this requires "DNS_UPSTREAM_QUERY_ENABLE"
```suggestion
 * Represents the count of queries, responses, failures handled by upstream DNS server.
 *
 * Requires `OPENTHREAD_CONFIG_DNS_UPSTREAM_QUERY_ENABLE`.
```